python/ray/train/v2/tests/test_checkpoint_manager.py

```diff
@@ -158,17 +160,76 @@ def test_before_init_train_context(tmp_path):
 
     # Assert without a checkpoint.
     assert checkpoint_manager.before_init_train_context(workers) == {
-        "checkpoint": [None] * 4
+        "checkpoint": [None] * 4,
+        "num_reported_checkpoints": [0] * 4,
     }
 
     # Assert with a checkpoint
     latest_checkpoint_result = _create_dummy_training_results(1, storage_context)[0]
-    checkpoint_manager._latest_checkpoint_result = latest_checkpoint_result
+    checkpoint_manager.register_checkpoint(latest_checkpoint_result)
     assert checkpoint_manager.before_init_train_context(workers) == {
-        "checkpoint": [latest_checkpoint_result.checkpoint] * 4
+        "checkpoint": [latest_checkpoint_result.checkpoint] * 4,
+        "num_reported_checkpoints": [1] * 4,
     }
 
 
+def test_before_init_train_context_from_actor(tmp_path):
+    # Create checkpoint manager actor
+    storage_context = StorageContext(
+        storage_path=tmp_path,
+        experiment_dir_name="my_experiment_name",
+    )
+    checkpoint_manager_actor_cls = ray.remote(CheckpointManager)
+    checkpoint_manager = checkpoint_manager_actor_cls.remote(
+        storage_context=storage_context,
+        checkpoint_config=CheckpointConfig(),
+    )
```

Comment:
Why's this an actor in this test?

Reply:
To verify that `before_init_train_context` passes the actor when it's called from the actor: https://github.com/ray-project/ray/blob/08b318c2b67cfdbe53c5a70243df8576f75b75ed/python/ray/train/v2/_internal/execution/checkpoint/checkpoint_manager.py#L315

Reply:
Hmm, I see. Maybe an assertion is better to avoid sending over an optional controller handle.

Also, I think this fully breaks the `RAY_TRAIN_RUN_CONTROLLER_AS_ACTOR` env var right? We may need to figure out how to remove that hack for Tune+Train for good.

Reply:
done.